### PR TITLE
docs: data-source documentation improvement

### PR DESCRIPTION
**Pull request by vtopalova23**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Refactoring (no functional changes, no api changes)
[X] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code

```
```

## What to Check

Verify that the following are valid

* ...

### CODE REVIEW

File: internal/provider/datasource_directory_entitlements.go [hunk 1]
```diff
@@ -47,10 +47,7 @@ func (ds *directoryEntitlementsDataSource) Schema(_ context.Context, _ datasourc
 
 To view all the resources that a directory and its subdirectories and subaccounts are entitled to use:
 * The directory must be a directory that is configured to manage its own entitlements.
-* You must be assigned to either the global account admin or global account viewers role.
-
-__Tip:__
```
Comment (vtopalova23) [inline internal/provider/datasource_directory_entitlements.go:52]: This tip looks to me like a duplication of the previous sentence.

File: internal/provider/datasource_directory.go [hunk 2]
```diff
@@ -64,13 +64,15 @@ __Further documentation:__
...
-					"\n\t\t 2. Monitor usage and costs on a directory level (costs only available for contracts that use the consumption-based commercial model)" +
-					"\n\t\t 3. Set custom properties and tags to the directory for identification and reporting purposes." +
-					"\n\t - `ENTITLEMENTS` Allows the assignment of a quota for services and applications to the directory from the global account quota for distribution to the subaccounts under this directory." +
-					"\n\t - `AUTHORIZATIONS` Allows the assignment of users as administrators or viewers of this directory. You must apply this feature in combination with the `ENTITLEMENTS` feature.",
+				MarkdownDescription: "The features that are enabled for the directory. Possible values are: \n" +
+					"\n  | value | description |" +
+					"\n  | ----- | ----------- |" +
+					"\n  | DEFAULT | All directories have the following basic feature enabled: " +
+					"<br> 1. Group and filter subaccounts for reports and filters " +
+					"<br> 2. Monitor usage and costs on a directory level (costs only available for contracts that use the consumption-based commercial model)" +
```
Comment (vtopalova23) [inline internal/provider/datasource_directory.go]: the <br> is to represent multi-line text in a single cell in a Markdown table.

**Comment by lechnerc77** [comment]
@vtopalova23 : Many thanks for this contribution! Highly appreciated, that makes the documentation much easier to consume

File: internal/provider/datasource_directory.go [hunk 3]
```diff
@@ -35,7 +35,7 @@ func (ds *directoryDataSource) Configure(_ context.Context, req datasource.Confi
 
 func (ds *directoryDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
 	resp.Schema = schema.Schema{
-		MarkdownDescription: `Get the details about a directory.
+		MarkdownDescription: `Gets the details about a directory.
```
Comment (vtopalova23) [inline internal/provider/datasource_directory.go:38]: Looking at the other provider's documentation, they use 3rd person singular for the data-source description.

File: docs/data-sources/directory.md [hunk 4]
```diff
@@ -37,33 +37,36 @@ data "btp_directory" "by_id" {
...
+  | CREATION_FAILED | The creation operation failed, and the entity was not created or was created but cannot be used. |
+  | UPDATING | Updating entity operation is in progress. |
+  | UPDATE_FAILED | The update operation failed, and the entity was not updated. |
+  | MOVING | Moving entity operation is in progress. |
+  | MOVE_FAILED | Entity could not be moved to a different location. |
+  | DELETING | Deleting entity operation is in progress. |
+  | DELETION_FAILED | The delete operation failed, and the entity was not deleted. |
+  | PENDING REVIEW | The processing operation has been stopped for reviewing and can be restarted by the operator. |
+  | MIGRATING | Migrating entity from NEO to CF. |
 - `subdomain` (String) Applies only to directories that have the user authorization management feature enabled. The subdomain is part of the path used to access the authorization tenant of the directory.
```
Comment (vtopalova23) [inline docs/data-sources/directory.md:72]: This is how it looks in the https://registry.terraform.io/tools/doc-preview:
Screenshot 2023-06-23 at 16 19 36

File: internal/provider/helper_doc_values_format.go [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package provider
+
+import (
+	"fmt"
+	"strings"
+)
+
+func getFormattedValueAsTableRow(val string, description string) string {
```
Comment (vtopalova23) [inline internal/provider/helper_doc_values_format.go:8]: I have introduced this function to reduce the ugly format of the table view

**Review by lechnerc77 — APPROVED** [review]
LGTM

**Pull request merged**